tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/relation/relation_one_to_one_test.dart

```diff
@@ -923,4 +924,151 @@ void main() {
       );
     }, skip: errors.isEmpty);
   });
+
+  testWithVariants(
+      'Given a class with an id relation,',
+      describeVariant: (v) =>
+          'when the class is${v.classIsServerOnly ? '' : ' not'} serverOnly'
+          ' and the id field is${v.nullableField ? '' : ' not'} nullable'
+          '${v.fieldScope == null ? '' : ' and the field scope is ${v.fieldScope!.name}'}'
+          ', then it should cause ${v.errors.length} error(s)',
```

Comment:
We discussed this on slack. It breaks a bit with the current pattern, but I hope you agree to it. 

BTW: Currently we do a lot of stuff in `group`s `body` callback. I'm not a fan of that as that callback is invoked during test setup, not on test run. This slow down the start of individual tests, and makes it harder to use break-points, as they may be hit before the test starts.